compiler/rustc_typeck/src/coherence/builtin.rs

```diff
@@ -76,12 +76,17 @@ fn visit_implementation_of_copy(tcx: TyCtxt<'_>, impl_did: LocalDefId) {
 
     debug!("visit_implementation_of_copy: self_type={:?} (free)", self_type);
 
+    let impl_ = match tcx.hir().expect_item(impl_did).kind {
+        ItemKind::Impl(hir::Impl { polarity: hir::ImplPolarity::Negative(_), .. }) => return,
+        ItemKind::Impl(impl_) => Some(impl_),
+        _ => None,
```

Comment:
is this branch even reachable? Can you change that to a `bug!` or `delay_span_bug`?